src/node_dotenv.cc

```diff
@@ -134,48 +137,69 @@ void Dotenv::ParseContent(const std::string_view input) {
 
   while (!content.empty()) {
     // Skip empty lines and comments
+    // Example:
+    //   # This is a comment
     if (content.front() == '\n' || content.front() == '#') {
-      auto newline = content.find('\n');
-      if (newline != std::string_view::npos) {
-        content.remove_prefix(newline + 1);
-        continue;
+      // Check if the first character of the content is a newline or a hash
+      if (content.front() == '\n') {
+        // If the first character is a newline, remove it
+        content.remove_prefix(1);
+      } else {
+        // If the first character is a hash, find the next newline character
+        auto newline = content.find('\n');
+        if (newline != std::string_view::npos) {
+          // If a newline is found, remove the comment line including the
+          // newline character.
+          content.remove_prefix(newline + 1);
+        }
```

Comment:
What's the point of differencing the case where there's a comment?

Reply:
If the first character is a newline, we remove the prefix of 1 from `content` because we're on length. In the case of a comment, however, the length must be calculated. 
So it's a small optimization

Reply:
Please add a test.